nucleus/dataset_item.py

```diff
@@ -237,3 +247,13 @@ def check_for_duplicate_reference_ids(dataset_items: Sequence[DatasetItem]):
         raise ValueError(
             f"Duplicate reference IDs found among dataset_items: {duplicates}"
         )
+
+
+def check_items_have_dimensions(dataset_items: Sequence[DatasetItem]):
+    for item in dataset_items:
+        has_width = getattr(item, "width")
+        has_height = getattr(item, "height")
+        if not (has_width and has_height):
+            raise Exception(
+                f"When using privacy mode, all items require a width and height. Missing for item: '{item.reference_id}'"
+            )
```

Comment:
fyi @gatli your script for privacy data upload will break after this change.

But i believe you can already get the image dimensions from the embedding service anyways?

Reply:
Thanks for the heads up 🙂